### PR TITLE
feat: Unix build

### PR DESCRIPTION
**Pull request by divy9881**
Signed-off-by: divy9881 <divy9881@gmail.com>

### CODE REVIEW

**Comment by hsluoyz** [comment]
@nodece @xcaptain @GopherJ please review.

**Comment by GopherJ** [comment]
Can we use cmake because I believe it's more universal, also it supports generating `compile_commands.json` which can be used directly by vim users who relies on for example https://github.com/clangd/coc-clangd

**Comment by divy9881** [comment]
CLion also supports Makefile and large projects like Bitcoin core protocol stack use makefile as a build tool, we had this discussion before, and we decided to go with Make.
Makefile is also universally used and is very light as compared to Cmake.

**Comment by divy9881** [comment]
And also you can generate `compile_commands.json` in Make

**Comment by divy9881** [comment]
@hsluoyz you can integrate travis CI with the casbin-cpp, I have added the config file.

**Comment by GopherJ** [comment]
@divy9881 Although I agree that Makefile is more light, but cmake is also not heavy~ In the company which I worker for, we always use CMake instead of Makefile.

But yes, let's start with a simple Makefile

**Comment by divy9881** [comment]
Yeah, I agree. If we had more complex build process we definitely would have used `CMake`. And also debian distribution of linux ships `make` tool in-built, so no hassle of setting up and installing cmake on the system.

**Comment by hsluoyz** [comment]
> @hsluoyz you can integrate travis CI with the casbin-cpp, I have added the config file.

Enabled: https://travis-ci.org/github/casbin/casbin-cpp

File: casbin/dllmain.cpp [hunk 1]
```diff
@@ -1,35 +0,0 @@
-/*
```
Comment (hsluoyz) [inline casbin/dllmain.cpp:1]: You deleted the DLL main function for Windows. Will Windows build still work?

Comment (divy9881) [reply casbin/dllmain.cpp:1]: Yes

**Comment by hsluoyz** [comment]
Why is binary file added to git?

image

File: casbin/dllmain.cpp (hunk 1 above)
Comment (divy9881) [reply casbin/dllmain.cpp:1]: The windows.h is a utility library related to windows specific implementation of C++. We don't have such specific dependencies in our application

Comment (divy9881) [reply casbin/dllmain.cpp:1]: They are actually redundant files.

**Comment by divy9881** [comment]
> Why is binary file added to git?
> 
> image

Okay, I will remove that file.

**Comment by hsluoyz** [comment]
Please fix all the warnings: https://dev.azure.com/Casbin/casbin/_build/results?buildId=173&view=results

image

File: casbin/dllmain.cpp (hunk 1 above)
Comment (hsluoyz) [reply casbin/dllmain.cpp:1]: So where is the DLL entry now?

**Comment by divy9881** [comment]
> Please fix all the warnings: https://dev.azure.com/Casbin/casbin/_build/results?buildId=173&view=results
> 
> image

This warnings can be fixed in another PR.

File: casbin/dllmain.cpp (hunk 1 above)
Comment (divy9881) [reply casbin/dllmain.cpp:1]: The type of library which we are using is a static library a .lib file for exporting this project and making use of this project in any other MVS C++ project. Hence, we don't require a DLL entry.

**Comment by hsluoyz** [comment]
> > Why is binary file added to git?
> > image
> 
> Okay, I will remove that file.

I didn't see it's removed. Also make sure this file doesn't show up in git history. Or I have to squash-merge the PR.

**Pull request merged**